lib/config/autoconfig.js

```diff
@@ -0,0 +1,367 @@
+        }.bind(this));
+
+        return newRegistry;
+    },
+
+    /**
+     * Lint SourceCodes against all configurations in the registry, and record results
+     *
+     * @param   {Object[]} sourceCodes  SourceCode objects for each filename
+     * @param   {Object}   config       ESLint config object
+     * @param   {progressCallback} [cb] Optional callback for reporting execution status
+     * @returns {Registry}              New registry with errorCount populated
+     */
+    lintSourceCode: function(sourceCodes, config, cb) {
+        var totalFilesLinting,
+            lintConfig,
+            lintResults,
+            ruleSets,
+            ruleSetIdx,
+            filenames,
+            lintedRegistry;
+
+        lintedRegistry = new Registry();
+        lintedRegistry.rules = assign({}, this.rules);
+        ruleSets = lintedRegistry.buildRuleSets();
+        lintedRegistry = lintedRegistry.stripExtraConfigs();
+
+        debug("Linting with all possible rule combinations");
+        filenames = Object.keys(sourceCodes);
+        totalFilesLinting = filenames.length * ruleSets.length;
+        filenames.forEach(function(filename) {
+            debug("Linting file: " + filename);
+            ruleSetIdx = 0;
+            ruleSets.forEach(function(ruleSet) {
+                lintConfig = assign({}, config, {rules: ruleSet});
+                try {
+                    lintResults = eslint.verify(sourceCodes[filename], lintConfig);
+                    lintResults.forEach(function(result) {
+                        lintedRegistry.rules[result.ruleId][ruleSetIdx].errorCount += 1;
+                    });
+                    // Deallocate for GC
+                    lintResults = null;
+                } catch (err) {
+                    console.error(err);
+                }
+                ruleSetIdx += 1;
+                if (cb) {
+                    cb(totalFilesLinting);  // eslint-disable-line callback-return
+                }
+            });
+            // Deallocate for GC
+            sourceCodes[filename] = null;
+        });
+
+        return lintedRegistry;
+    }
+};
+
+/**
+ * Extract rule configuration into eslint:recommended where possible.
+ *
+ * This will return a new config with `"extends": "eslint:recommended"` and
+ * only the rules which have configurations different from the recommended config.
+ *
+ * @param   {Object} config config object
+ * @returns {Object}        config object using `"extends": "eslint:recommended"`
+ */
+function extendFromRecommended(config) {
+    var newConfig = assign({}, config);
+    var recRules = Object.keys(recConfig.rules).filter(function(ruleId) {
+        return (recConfig.rules[ruleId] === 2 || recConfig.rules[ruleId][0] === 2);
+    });
+
+    recRules.forEach(function(ruleId) {
+        if (isEqual(recConfig.rules[ruleId], newConfig.rules[ruleId])) {
+            delete newConfig.rules[ruleId];
+        }
+    });
+    newConfig.extends = "eslint:recommended";
```

Comment:
Sorry, I don't understand what you are suggesting.  What value would you put up top, and what is the problem you noticed in the code that doing so would solve?